mitosheet/mitosheet/ai/ai_utils.py

```diff
@@ -40,17 +47,25 @@ def get_code_string_from_last_expression(code: str, last_expression: ast.stmt) -
     return "\n".join(relevant_lines)
 
 def fix_final_dataframe_name(code: str, new_df_name: str, is_series: bool) -> str:
-    ast_before = ast.parse(code)
-    last_expression = ast_before.body[-1]
-
-    last_expression_string = get_code_string_from_last_expression(code, last_expression)
+    
+    last_expression_string = get_code_string_from_last_expression(code)
 
+    if last_expression_string is None:
+        return code
+    
+    # We only want to replace the last instance, so we have to 
+    # reverse the string, replace the dataframe name with the reverse, and then reverse it back again
```

Comment:
This comment should probably be moved down to the `replace_last_instance_in_string` function